controllers/volumereplicationgroup_controller.go

```diff
@@ -398,6 +404,95 @@ func (v *VRGInstance) validateVRGState() error {
 	return nil
 }
 
+func (v *VRGInstance) restorePVs() error {
+	// TODO: refactor this per this comment: https://github.com/RamenDR/ramen/pull/197#discussion_r687246692
+	vrgCondition := findCondition(v.instance.Status.Conditions, PVConditionMetadataAvailable)
+	if vrgCondition != nil && vrgCondition.Status == metav1.ConditionTrue {
+		v.log.Info("VRGConditionAvailable found. PV restore must have already been applied")
+
+		return nil
+	}
+
+	msg := "Restoring PV metadata"
+	setPVMetadataProgressingCondition(&v.instance.Status.Conditions, v.instance.Generation, msg)
+
+	v.log.Info(fmt.Sprintf("Restoring PVs to this managed cluster %v", v.instance.Spec.S3ProfileList))
+	// TODO: this is added in the last minute while merging.  More thought will be put in this
+	for _, s3ProfileName := range v.instance.Spec.S3ProfileList {
+		pvList, err := v.listPVsFromS3Store(s3ProfileName)
+		if err != nil {
+			return fmt.Errorf("failed to retrieve PVs from S3 store (%w)", err)
+		}
+
+		v.log.Info(fmt.Sprintf("Found %d PVs", len(pvList)))
+
+		if len(pvList) == 0 {
```

Comment:
Skip this check, if there are no PVs then we still need to mark the final MetadataCondition as true.